File: missingFeature/models/NotificationModel.js
import mongoose from 'mongoose';

const { Schema, model } = mongoose;

const notificationSchema = new Schema(
  {
    user: {
      type: Schema.Types.ObjectId,
      ref: "user",
    },
    guestToken: {
      type: String,
    },
    title: {
      type: String,
      required: true,
    },
    body: {
      type: String,
      required: true,
    },
    caseID: {
      type: String,
      required: true,
    },
    isRead: {
      type: Boolean,
      default: false,
    },
  },
  {
    timestamps: true,
  }
);

notificationSchema.statics.deleteOldNotifications = async function () {
  const fifteenSecondsAgo = new Date();
  fifteenSecondsAgo.setSeconds(fifteenSecondsAgo.getSeconds() - 86400);

  await this.deleteMany({ createdAt: { $lt: fifteenSecondsAgo } });
};

const Notification = model("notification", notificationSchema);

export default Notification;

File: Frontend/missingpersonapp/lib/features/Notifications/models/notification_model.dart
class NotificationModel {
  final String id;
  final String title;
  final String body;
  final String caseID;
  final bool isRead;
  final DateTime createdAt;

  NotificationModel({
    required this.id,
    required this.title,
    required this.body,
    required this.caseID,
    required this.isRead,
    required this.createdAt,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['_id'],
      title: json['title'],
      body: json['body'],
      caseID: json['caseID'],
      isRead: json['isRead'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}
